File: web-sources/S8-fwks-palm/boxes/b2/B2Box.js

import { S8 } from "/S8-api/S8Context.js";
import { S8Object } from "/S8-api/S8Object.js";

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";




/**
 * 
 */
S8.page.CSS_import('/S8-fwks-palm/boxes/b2/B2Box.css');


/**
 * 
 */
export class B2Box extends S8Object {

    /** @type {HTMLDivElement} */
    wrapperNode;

    elements;

    /**
     * 
     */
    constructor() {
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("b2box");
    }


    /**
     * 
     * @returns 
     */
    getEnvelope() {
        return this.wrapperNode;
    }

    /** @returns{S8Object[]} */
    getComponents() {
        return this.elements;
    }

    /** @param {S8Object[]} elements */
    S8_set_elements(elements) {
        this.elements = elements;
        S8WebFront.setS8Elements(this.wrapperNode, elements);
    }


    /** render */
    S8_render() { }


    /** dispose */
    S8_dispose() { }

}

File: sources/com/s8/pkgs/palm/components/boxes/b2/B2Box.java
package com.s8.pkgs.palm.components.boxes.b2;

import com.s8.api.web.S8WebFront;
import com.s8.api.web.S8WebObject;
import com.s8.pkgs.palm.components.PalmWebSources;
import com.s8.pkgs.ui.carbide.HTML_S8WebNode;


/**
 * 
 * @author pierreconvert
 *
 */
public class B2Box extends HTML_S8WebNode {
	
	
	public final static String WEBPATH = PalmWebSources.ROOT_WEBPATH + "/boxes/b2/B2Box";
	
	

	
	/**
	 * 
	 * @param front
	 */
	
	public B2Box(S8WebFront front) {
		super(front, WEBPATH);
	}

	
	/**
	 * 
	 * @param items
	 */
	public void setElements(S8WebObject... elements) {
		vertex.outbound().setObjectListField("elements", elements);
	}
	
}
